staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -335,7 +335,7 @@ func (a *mutatingDispatcher) callAttrMutatingHook(ctx context.Context, h *admiss
 	}
 	patchObj, err := jsonpatch.DecodePatch(result.Patch)
 	if err != nil {
-		return false, apierrors.NewInternalError(err)
+		return false, apierrors.NewInternalError(fmt.Errorf("failed to decode patch returned by webhook %q: %w", h.Name, err))
```

Comment:
I'm actually wondering if we should switch this error to an `ErrCallingWebhook` type as we return above if `VerifyAdmissionResponse` fails. That already includes the name, and would trigger the `failurePolicy` of the webhook if it returned a malformed patch

Reply:
yes, I think that would be better. Then if failurepolicy is ignore, the errors won't pollute audit log with 500s in the first place which was my main motivation for this PR. i'll add a release note since this has some implications for metrics also.